docs/EVE-IMAGE-SOURCES.md

```diff
@@ -0,0 +1,616 @@
+```
+
+The results of the inspection will yield information about the container image, including tags that tell which git commit
+was used when building this image. In the above example:
+
+```json
+      "Labels": {
+        "org.mobyproject.config": "{\"capabilities\":[\"all\"],\"binds\":[\"/lib/modules:/lib/modules\",\"/dev:/dev\",\"/etc/resolv.conf:/etc/resolv.conf\",\"/r
+un:/run\",\"/config:/config\",\"/:/hostfs\",\"/persist:/persist:rshared,rbind\",\"/usr/bin/containerd:/usr/bin/containerd\"],\"devices\":[{\"path\":\"all\",\"ty
+pe\":\"a\",\"major\":0,\"minor\":0}],\"net\":\"host\",\"pid\":\"host\",\"rootfsPropagation\":\"shared\"}",
+        "org.mobyproject.linuxkit.revision": "unknown",
+        "org.mobyproject.linuxkit.version": "unknown",
+        "org.opencontainers.image.revision": "12a6fb8d29a6b47f998cd077dfb92213f0e6a55f",
+        "org.opencontainers.image.source": "https://github.com/linuxkit/linuxkit"
+      },
+```
+
+The git commit for the above source is given in the label `"org.opencontainers.image.revision": "12a6fb8d29a6b47f998cd077dfb92213f0e6a55f"`, i.e.
+you can get the same commit via `git checkout 12a6fb8d29a6b47f998cd077dfb92213f0e6a55f`.
+
+The above _should_ be the same as the current git commit.
+
+Further, you can validate that the current contents of `pkg/pillar` give the correct output, by running:
+
+```sh
+$ linuxkit pkg show-tag pkg/pillar
+lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b
+```
+
+The above `17837a9fcd05c765e9a1f6707b2e48f0f1dd215b` is the same git tree hash used in the container image tag,
+`lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b-amd64`.
+
+## Container sources to Content
+
+With the specific source directory and commit in hand for each container image, we turn to the various sources that are used.
+
+There are several types of content that are loaded used to create the container image.
+
+### Local content
+
+Local content includes anything in the same directory, which therefore is used in the
+[docker build context](https://docs.docker.com/engine/reference/commandline/build/) when creating the container image.
+
+Everything used in the source directory is committed to git. This includes sources, the `Dockerfile` itself, and the `build.yaml`
+file used to control the image build via `linuxkit pkg build`.
+
+Further, `linuxkit pkg build` is used, it validates that the git repository is "clean", i.e. that there are no uncommitted changes
+or files not saved in git. If there are, it treats the repository as "dirty", and appends both `dirty` and a hash of all of the file
+contents in the directory to the tag. Thus `lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b-amd64` means that it checked,
+and there were no uncommitted changes or files in the source directory. If there were, it would have indicated on the tag as
+`lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b-dirty-<hash_of_file_contents>-amd64`.
```

Comment:
I think one can make the statement that if -dirty is included it is not possible to know the complete source, since some files very modified locally in the workspace where the image was built. Should we clarify that in the document?

Reply:
Yeah, good point. You _might_ get lucky and have the same files - if the hash works out, i.e. `lkt pkg show-tag` gives the same result - but you might not. There is no way to "get to it", like `git checkout`. Will add.